### PR TITLE
Refactor theme system to apply them earlier in page load

### LINKED ISSUE #2242: [BUG] Dark-mode switch on refresh should be faster
Having dark-mode as your default, should not keep you in the light as much as the website does now when you refresh the page/visit a link.

(asked a while back, but forgot about it, so now submitting this issue to track)

### PR DESCRIPTION
**Pull request by apmorton**
Fixes #2242

The root problem causing delayed theme application is that `Themer` is part of our main webpack bundle, and it executes dead last.
`client-options.js` is huge and takes a long time to download/execute, along with our actual vendor/main bundles.  Fixing that is a whole other can of worms.

However, a simplerish solution is to hoist the theme css out of our webpack bundle, and apply it from a small inline script as soon as possible.

In order to do this (and not have it be a complex mess), all theme css needs to be part of the main css bundle and be applied as-needed depending on which theme is selected.

The simplest thing is to make every theme related style depend on an attribute on the html tag:
```html
<html data-theme="dark">
```

css doesn't allow true nesting, but sass does:
```scss
html[data-theme=dark] {
   // many rules here
}
```

This makes the change fairly trivial:
- convert all our existing styles to sass (which is a true superset of css, so just rename the files)
- enable sass loader in webpack
- chuck a small inline script into the pug layout to apply the theme if possible

### CODE REVIEW

**Comment by partouf** [comment]
Am I correct to assume this definitively breaks IE6 support?

**Comment by apmorton** [comment]
_IE6?!_ IE11 barely works

**Comment by partouf** [comment]
> _IE6?!_ IE11 barely works

Oh pf yeah that never worked, sorry, oops. Does this still work in IE11 then?

**Comment by apmorton** [comment]
Untested in IE11, but I don't think any of the changes made should break it.

**Comment by partouf** [comment]
Cool, I'll test

**Comment by partouf** [comment]
This somehow broke the new library stars
image

Not sure why. Investigating.

**Comment by partouf** [comment]
Ignore me, Totally wrong, I forgot how the thing that I made actually works

**Comment by partouf** [comment]
IE is already broken on prod:

```
SCRIPT1005: Expected '('
vendor.v3.08a4d3809e3d6e07e6be.js (2,898746)
```

Which is in this bit in ./node_modules/yallist/iterator.js :
```
'use strict'
module.exports = function (Yallist) {
  Yallist.prototype[Symbol.iterator] = function* () {
    for (let walker = this.head; walker; walker = walker.next) {
      yield walker.value
    }
  }
}
```

So nevermind!

**Review by partouf — APPROVED** [review]
Works great!

**Comment by mattgodbolt** [comment]
> IE is already broken on prod:
> 
> ```
> SCRIPT1005: Expected '('
> vendor.v3.08a4d3809e3d6e07e6be.js (2,898746)
> ```
> 
> Which is in this bit in ./node_modules/yallist/iterator.js :
> 
> ```
> 'use strict'
> module.exports = function (Yallist) {
>   Yallist.prototype[Symbol.iterator] = function* () {
>     for (let walker = this.head; walker; walker = walker.next) {
>       yield walker.value
>     }
>   }
> }
> ```
> 
> So nevermind!

Oh yikes...I guess.... this means we can probably start using newer JS in the client side? Maybe...?

File: views/_layout.pug [hunk 1]
```diff
@@ -9,7 +9,18 @@ html(lang="en")
     block styles
         link(href=require("vendor.css") rel="stylesheet")
         link(href=require("main.css") rel="stylesheet")
-        style#theme
+
+    //- partially apply theme as soon as possible to minimize the page load flashbang duration when using dark theme
```
Comment (mattgodbolt) [inline views/_layout.pug:13]: Love the "flashbang" terminology!

**Pull request merged**